### PR TITLE
Add 'move between shares' acceptance test

### PR DESCRIPTION
**Pull request by phil-davis**
his is an extra test scenario on top of PR #1018

### CODE REVIEW

File: tests/acceptance/features/apiActivity/list.feature [hunk 1]
```diff
@@ -1024,3 +1024,47 @@ Feature: List activity
...
+      | typeicon         | /^icon-move/                                                                                                                                                                                                                                                                                                                      |
+      | subject_prepared | /^You moved <file link=\"%base_url%\/(index\.php\/)?apps\/files\/\?dir=\/FolderForBrian\&scrollto=textfile0\.txt\" id=\"\">FolderForBrian\/textfile0\.txt<\/file> to <file link=\"%base_url%\/(index\.php\/)?apps\/files\/\?dir=\/FolderForCarol\&scrollto=textfile0\.txt\" id=\"\d+\">FolderForCarol\/textfile0\.txt<\/file>/ |
+    And the activity number 1 of user "Carol" should match these properties:
+      | type             | /^file_moved/                                                                                                                                                                                                        |
```
Comment (phil-davis) [inline tests/acceptance/features/apiActivity/list.feature:1054]:
```suggestion
      | type             | /^file_created/                                                                                                                                                                                                        |
```

File: tests/acceptance/features/apiActivity/list.feature [hunk 2]
```diff
@@ -1024,3 +1024,47 @@ Feature: List activity
...
+      | user             | /^Alice/                                                                                                                                                                                                               |
+      | affecteduser     | /^Carol/                                                                                                                                                                                                              |
+      | app              | /^files$/                                                                                                                                                                                                              |
+      | subject          | /^moved_by/                                                                                                                                                                                                          |
```
Comment (phil-davis) [inline tests/acceptance/features/apiActivity/list.feature:1058]:
```suggestion
      | subject          | /^created_by/                                                                                                                                                                                                          |
```

File: tests/acceptance/features/apiActivity/list.feature [hunk 3]
```diff
@@ -1024,3 +1024,47 @@ Feature: List activity
...
+      | object_name      | /^\/FolderForCarol\/textfile0\.txt/                                                                                                                                                                                            |
+      | object_type      | /^files$/                                                                                                                                                                                                              |
+      | typeicon         | /^icon-move/                                                                                                                                                                                                      |
```
Comment (phil-davis) [inline tests/acceptance/features/apiActivity/list.feature:1061]:
```suggestion
      | typeicon         | /^icon-add-color/                                                                                                                                                                                                      |
```

File: tests/acceptance/features/apiActivity/list.feature [hunk 4]
```diff
@@ -1024,3 +1024,47 @@ Feature: List activity
...
+      | subject_prepared | /^<user display-name=\"Alice Hansen\">Alice<\/user> created <file link=\"%base_url%\/(index\.php\/)?apps\/files\/\?dir=\/FolderForCarol\&scrollto=textfile0\.txt\" id=\"\d+\">FolderForCarol\/textfile0\.txt<\/file>$/ |
```
Comment (phil-davis) [inline tests/acceptance/features/apiActivity/list.feature:1062]: The subject_prepared is actually text about Alice moved file... and includes information about where the file was moved from (the folder name `FolderForBrian` is exposed in the text)

IMO the behavior is not correct???

@JammingBen

File: tests/acceptance/features/apiActivity/list.feature [hunk 5]
```diff
@@ -1024,3 +1024,57 @@ Feature: List activity
...
+      | object_type      | /^files$/                                                                                                                                                                                                                                                                                                                         |
+      | typeicon         | /^icon-move/                                                                                                                                                                                                                                                                                                                      |
+      | subject_prepared | /^You moved <file link=\"%base_url%\/(index\.php\/)?apps\/files\/\?dir=\/FolderForBrian\&scrollto=textfile0\.txt\" id=\"\">FolderForBrian\/textfile0\.txt<\/file> to <file link=\"%base_url%\/(index\.php\/)?apps\/files\/\?dir=\/FolderForCarol\&scrollto=textfile0\.txt\" id=\"\d+\">FolderForCarol\/textfile0\.txt<\/file>/ |
+    And the activity number 1 of user "Brian" should match these properties:
```
Comment (phil-davis) [inline tests/acceptance/features/apiActivity/list.feature:1053]: This step passes - good.
Brian sees a delete, and the destination folder of the "real move" is not exposed to Brian.

File: tests/acceptance/features/apiActivity/list.feature (hunk 4 above)
Comment (JammingBen) [reply tests/acceptance/features/apiActivity/list.feature:1072]: Ohh I never thought of this scenario tbh. This looks bad - I need to check.

Comment (phil-davis) [reply tests/acceptance/features/apiActivity/list.feature:1072]: There could be whole groups of users on each side of a move between shares. One group sees a "delete"  and the other group sees a "create"

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiActivity-master-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/activity/2279/25/1

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/activity/2279/25/9
```
| subject_prepared | /^<user display-name=\"Alice Hansen\">Alice<\/user> created <file link=\"%base_url%\/(index\.php\/)?apps\/files\/\?dir=\/FolderForCarol\&scrollto=textfile0\.txt\" id=\"\d+\">FolderForCarol\/textfile0\.txt<\/file>$/ |

'/^<user display-name=\"Alice Hansen\">Alice<\/user> created <file link=\"http\:\/\/server\/(index\.php\/)?apps\/files\/\?dir=\/FolderForCarol\&scrollto=textfile0\.txt\" id=\"\d+\">FolderForCarol\/textfile0\.txt<\/file>$/' 
      does not match 
'<user display-name="Alice Hansen">Alice</user> moved <file link="http://server/apps/files/?dir=/FolderForBrian&scrollto=textfile0.txt" id="">FolderForBrian/textfile0.txt</file> to <file link="http://server/apps/files/?dir=/FolderForCarol&scrollto=textfile0.txt" id="2147484218">FolderForCarol/textfile0.txt</file>'

```

**Comment by JammingBen** [comment]
@phil-davis I updated the source branch, moving files between shares should display correct activity messages now.

**Comment by JammingBen** [comment]
Updated the source branch here.

**Comment by JammingBen** [comment]
Looking good. @phil-davis Any more scenarios you want to add with this PR?

In the future we could expand the tests so that not only Alice (= the share owner), but the share recipients make rename/move-actions. But that's not a must for now IMO.

**Comment by phil-davis** [comment]
> Any more scenarios you want to add with this PR?

I added "Scenario: move a file between shares shared to different groups"

That checks that the users get the expected activity entries when they are members of a group, and that more than one member of a group gets the activity entry. It passes.

That's enough for now. Group activities basically work. We can merge this into `activity-rename-move` when CI finishes, and think about more test scenarios later.

**Pull request merged**